### ISSUE TITLE
Ability to Configure Dataverse Facets

### ISSUE THREAD

**Issue by shlake** [IQSS/dataverse#2715]
Is there any way to have an option to turn off showing datasets at the dataverse level so the default is dataverse view only, like this:

![dv-only](https://cloud.githubusercontent.com/assets/3809325/10886949/79a095fe-8152-11e5-958d-f35f8b58e6b5.png)

Use case is to create a dataverse that only allows creation of sub-dataverses (and those dataverses, only allow datasets). Since there will be no datasets at this upper dataverse level, we do not want to see the datasets in the sub-dataverses.

I guess this would be a dataverse facet selection?

**Comment by pdurbin** [comment]
Hi @shlake as I mentioned on [the thread](https://groups.google.com/forum/#!topic/dataverse-community/i1aBc-XuFh4) on the mailing list, is it currently not possible to only show dataverses by default in "browse" mode.

https://github.com/IQSS/dataverse/blob/v4.2/src/main/java/edu/harvard/iq/dataverse/search/SearchIncludeFragment.java#L207 is the area of the code a developer would look at.

Basically, are are two modes, "browse" and "search". By default, we hide files in browse mode:


|             | Dataverses | Datasets | Files  |
| ---------- | --------------- | ----------- | ------- |
| Browse | shown         | shown   | hidden |
| Search | shown          | shown   | shown |

It sounds like what you'd like is for datasets to be hidden in browse mode for certain dataverses. That is to say, if would be configurable per dataverse. Please correct me if I misunderstand!

I'm going to assign this to @eaquigley to think about from a usability/feature perspective.

**Comment by shlake** [comment]
Yes, your interpretation is correct, "datasets to be hidden in browse mode for certain dataverses" and configurable per dataverse.

**Comment by pdurbin** [comment]
@shlake are you still interested in this?

**Comment by shlake** [comment]
@pdurbin Thanks for checking in. The case I described above is not needed at UVA (being able to only see dataverses at the top level). Since I created this ticket, we have turned on dataset creation at the top level. I put all the dataverses in the "Featured" carousel at the top.

But with the plan on having a customized "home" page, I would think being able to set browse views (picking and choosing facets)  on the home page would be part of a customizable home page?

**Comment by pdurbin** [comment]
@shlake interesting. Yeah, we have dataset creation turned on at the top level too.

No, the idea with the customized home page being developed in #3639 is that is a static HTML page (though you could make it more dynamic with JavaScript) that you have complete control over. The code to support this is in pretty good shape but we'd like to provide sample HTML files that are useful to people beyond "This is a custom home page" and "This is a custom footer". Here's a screenshot of the documentation so far, which you can also see at http://guides.dataverse.org/en/3639-homepage-customization/installation/config.html#branding-your-installation

![screen shot 2017-06-09 at 9 38 49 am](https://user-images.githubusercontent.com/21006/26978095-2c3ad16c-4cf8-11e7-8267-a2cc99151d10.png)

**Comment by pdurbin** [comment]
> The case I described above is not needed at UVA (being able to only see dataverses at the top level). Since I created this ticket, we have turned on dataset creation at the top level. I put all the dataverses in the "Featured" carousel at the top.

I'm closing this issue based on the comment by @shlake above.

**Comment by vkush** [comment]
Dear @pdurbin , could it be possible to consider this option from the issue again (to turn off showing datasets at the dataverse level)? To have a possibility to configure, what combination will be shown per default (dataverse-dataset-files) - maybe like an edit option of every dataverse (via API [and GUI]).
It will be practical for the cases, where dataverses are collecting huge amount of datasets, which are only "timestamps" of something. When you need scrolling only on dataverses, and only after that you go inside the needed dataverse to see all these timestamps (datasets).

Now in the root-URL per default we have .../root?q=&types=dataverses%3Adatasets&sort=dateSort&order=desc&page=1
And to switch off datasets is possible to manually edit URL to .../root?q=&types=dataverses&sort=dateSort&order=desc&page=1 . Such a behaviour would be nice to have as a config parameter.

**Comment by pdurbin** [comment]
@vkush thanks for the feeback. I plugged those URLs in our demo server to make sure I'm understanding what you want.

https://demo.dataverse.org/dataverse/demo?q=&types=dataverses%3Adatasets&sort=dateSort&order=desc&page=1 is the current, unconfigurable experience of collections ("dataverses") always being shown:

![Screen Shot 2023-01-27 at 2 23 12 PM](https://user-images.githubusercontent.com/21006/215179807-09e1d18c-122a-4813-9f06-16fb308d84a0.png)

https://demo.dataverse.org/dataverse/demo?q=&types=dataverses&sort=dateSort&order=desc&page=1 shows what you want, the ability to not show datasets, like this:

![Screen Shot 2023-01-27 at 2 24 00 PM](https://user-images.githubusercontent.com/21006/215179806-cf7ce52e-2a51-43c3-b9ae-f320f98893ac.png)

It does appear that this is exactly what @shlake opened this issue about. However, since she's no longer interested in it, my (slight) preference would be for you to open a fresh issue. If this sounds like too much work, please let me (or @shlake ) know and we can open it back up.

Another thing, as you may have heard, we are planning a big rewrite of the front end ( https://dataverse.org/blog/restructuring-dataverse-ui-single-page-application ) so issues like this feel a little odd these days. That is, I'd personally rather put effort into the new front end instead of code we're going to ultimately throw away. I hope that makes sense. 😄 

Have a nice weekend!
